.github/workflows/validate-rules-syntax.yaml

```diff
@@ -0,0 +1,100 @@
+on: pull_request
+permissions:
+  contents: read
+  pull-requests: read
+jobs:
+  validate_loki_recording_rules:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v2
+
+      - name: Create tmp dir 
+        run: mkdir -p validation/recording-rules
+
```

Comment:
Consider a step like `kustomize build . | kustomize cfg grep kind=PrometheusRule | kustomize cfg merge validation/recording-rules`

This would not only test that the kustomization is valid; but it would apply any patches we decide to include in the repository.